Mathlib/RingTheory/IntegrallyClosed.lean

```diff
@@ -11,102 +11,153 @@ import Mathlib.RingTheory.Localization.Integral
 /-!
 # Integrally closed rings
 
-An integrally closed domain `R` contains all the elements of `Frac(R)` that are
-integral over `R`. A special case of integrally closed domains are the Dedekind domains.
+An integrally closed ring `R` in an algebra `A` contains all the elements of `A` that are
+integral over `R`. A special case of integrally closed rings are the Dedekind domains.
+[Stacks: integrally closed ring](https://stacks.math.columbia.edu/tag/00GP)
 
 ## Main definitions
 
-* `IsIntegrallyClosed R` states `R` contains all integral elements of `Frac(R)`
+* `IsIntegrallyClosed R A` states `R` contains all integral elements of `A`
 
 ## Main results
 
-* `isIntegrallyClosed_iff K`, where `K` is a fraction field of `R`, states `R`
+* `AlgEquiv.isIntegrallyClosed_iff e`, where `e` is an isomorphism of `R`-algebras `A` and `A'`,
+  states `R` is integrally closed in `A` iff it is integrally closed in `A'`.
+* `isIntegrallyClosed_fractionRing_iff K`, where `K` is a fraction field of `R`, states `R`
   is integrally closed iff it is the integral closure of `R` in `K`
+
+## TODO: Related notions
+
+The following definitions are closely related, especially in their applications in Mathlib.
+
+A *normal domain* is a domain that is integrally closed in its field of fractions.
+[Stacks: normal domain](https://stacks.math.columbia.edu/tag/0309)
+Normal domains are the major usecase of `IsIntegrallyClosed` at the time of writing, and we have
+quite a few results that can be moved wholesale to a new `NormalDomain` definition.
+In fact, before PR #6126 `IsIntegrallyClosed` was exactly defined to be a normal domain.
+(So you might want to copy some of its API when you define normal domains.)
+
+A normal ring means that localizations at all prime ideals are normal domains.
+[Stacks: normal ring](https://stacks.math.columbia.edu/tag/00GV)
+This implies integral closedness in its `FractionRing`,
+[Stacks: Lemma 10.37.12](https://stacks.math.columbia.edu/tag/034M)
+but is equivalent to it only under some conditions (reduced + finitely many minimal primes),
+[Stacks: Lemma 10.37.16](https://stacks.math.columbia.edu/tag/034M)
```

Comment:
I think links that jump to the def/lemmas *within* the section should be preferred (though these aren't perfect yet as the def/lemmas get covered beneath the header). Also, referring to the tag should be more robust, as the numbering may change.
```suggestion
[Stacks: normal domain](https://stacks.math.columbia.edu/tag/037B#0309)
Normal domains are the major use case of `IsIntegrallyClosed` at the time of writing, and we have
quite a few results that can be moved wholesale to a new `NormalDomain` definition.
In fact, before PR #6126 `IsIntegrallyClosed` was exactly defined to be a normal domain.
(So you might want to copy some of its API when you define normal domains.)

A normal ring means that localizations at all prime ideals are normal domains.
[Stacks: normal ring](https://stacks.math.columbia.edu/tag/037B#00GV)
This implies integral closedness in its `FractionRing`,
[Stacks: Tag 034M](https://stacks.math.columbia.edu/tag/037B#034M)
but is equivalent to it only under some conditions (reduced + finitely many minimal primes),
[Stacks: Tag 030C](https://stacks.math.columbia.edu/tag/037B#030C)
```